src/main/java/edu/wpi/grip/ui/pipeline/input/ColorInputSocketView.java

```diff
@@ -0,0 +1,70 @@
+package edu.wpi.grip.ui.pipeline.input;
+
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.value.ChangeListener;
+import javafx.scene.control.ColorPicker;
+import javafx.scene.paint.Color;
+import org.bytedeco.javacpp.opencv_core.Scalar;
+
+/**
+ * An {@link InputSocketView} that lets the user set the value of a {@link Scalar} with a {@link ColorPicker}
+ */
+public class ColorInputSocketView extends InputSocketView<Scalar> {
+
+    private final ColorPicker colorPicker;
+    private final ChangeListener<Color> updateSocketFromColor;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Perhaps 'updateSocketFromColor' could be replaced by a local variable.](https://www.codacy.com/app/WPIRoboticsProjects/GRIP/file/1275184114/issues/source?bid=2943107&fileBranchId=2943116#l19)